File: web/src/lib/nutrition.ts
import type { TreatKind } from './types';

export interface LaneMeta {
  kind: TreatKind;
  /** Was der Eintrag bedeutet */
  title: string;
  /** Kurzform für enge Stellen */
  short: string;
  emoji: string;
  color: string;
  tagline: string;
  /** Beschriftung des Plus-Knopfes */
  add: string;
}

/**
 * Zwei Spuren, beide negativ: eingetragen wird, was daneben ging. Der Ton
 * bleibt trotzdem sachlich — ein Eintrag ist Buchhaltung, keine Strafpredigt.
 */
export const LANES: Record<TreatKind, LaneMeta> = {
  sweets: {
    kind: 'sweets',
    title: 'Süßes gegessen',
    short: 'Gegessen',
    emoji: '🍫',
    color: 'var(--color-cocoa)',
    tagline: 'Schokolade, Kuchen, Chips — alles, was zwischendurch reinrutscht.',
    add: 'Genascht',
  },
  drinks: {
    kind: 'drinks',
    title: 'Süßes getrunken',
    short: 'Getrunken',
    emoji: '🥤',
    color: 'var(--color-mint)',
    tagline: 'Limo, Saft, Zucker im Kaffee — alles außer Wasser und Tee.',
    add: 'Getrunken',
  },
};

export const LANE_LIST = [LANES.sweets, LANES.drinks];

File: app/lib/core/nutrition.dart
import 'package:flutter/material.dart';

import '../theme.dart';
import 'types.dart';

class LaneMeta {
  const LaneMeta({
    required this.kind,
    required this.title,
    required this.short,
    required this.emoji,
    required this.color,
    required this.tagline,
    required this.add,
  });

  final TreatKind kind;

  /// Was der Eintrag bedeutet
  final String title;

  /// Kurzform für enge Stellen
  final String short;
  final String emoji;
  final Color color;
  final String tagline;

  /// Beschriftung des Plus-Knopfes
  final String add;
}

/// Zwei Spuren, beide negativ: eingetragen wird, was danebenging. Der Ton
/// bleibt trotzdem sachlich — ein Eintrag ist Buchhaltung, keine Strafpredigt.
const _sweets = LaneMeta(
  kind: TreatKind.sweets,
  title: 'Süßes gegessen',
  short: 'Gegessen',
  emoji: '🍫',
  color: C.cocoa,
  tagline: 'Schokolade, Kuchen, Chips — alles, was zwischendurch reinrutscht.',
  add: 'Genascht',
);

const _drinks = LaneMeta(
  kind: TreatKind.drinks,
  title: 'Süßes getrunken',
  short: 'Getrunken',
  emoji: '🥤',
  color: C.mint,
  tagline: 'Limo, Saft, Zucker im Kaffee — alles außer Wasser und Tee.',
  add: 'Getrunken',
);

const lanes = <TreatKind, LaneMeta>{
  TreatKind.sweets: _sweets,
  TreatKind.drinks: _drinks,
};

const laneList = [_sweets, _drinks];
